src/LocalResourceProvider.ts

```diff
@@ -0,0 +1,97 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { AzExtParentTreeItem, AzExtTreeItem, callWithTelemetryAndErrorHandling, IActionContext } from "@microsoft/vscode-azext-utils";
+import * as path from 'path';
+import { Disposable, workspace, WorkspaceFolder } from "vscode";
+import { LocalResourceProvider } from "./api";
+import { tryGetFunctionProjectRoot } from "./commands/createNewProject/verifyIsProject";
+import { getFunctionAppName, getJavaDebugSubpath } from "./commands/initProjectForVSCode/InitVSCodeStep/JavaInitVSCodeStep";
+import { funcVersionSetting, JavaBuildTool, javaBuildTool, ProjectLanguage, projectLanguageSetting } from "./constants";
+import { FuncVersion, tryParseFuncVersion } from "./FuncVersion";
+import { localize } from "./localize";
+import { InitLocalProjectTreeItem } from "./tree/localProject/InitLocalProjectTreeItem";
+import { InvalidLocalProjectTreeItem } from "./tree/localProject/InvalidLocalProjectTreeItem";
+import { LocalProjectTreeItem } from "./tree/localProject/LocalProjectTreeItem";
+import { dotnetUtils } from "./utils/dotnetUtils";
+import { getWorkspaceSetting } from "./vsCodeConfig/settings";
+
+export class FunctionsLocalResourceProvider implements LocalResourceProvider {
+
+    public disposables: Disposable[] = [];
+
+    public async provideResources(parent: AzExtParentTreeItem): Promise<AzExtTreeItem[] | null | undefined> {
+
+        return await callWithTelemetryAndErrorHandling('AzureAccountTreeItemWithProjects.provideResources', async (context: IActionContext) => {
```

Comment:
Are you able to remove `AzureAccountTreeItemWithProjects` now?  That's where this implementation is from, right?

Reply:
We can, but lets do it later since the tests rely on it... 😭